test/lit/passes/remove-unused-module-elements-refs.wast

```diff
@@ -1,33 +1,39 @@
 ;; NOTE: Assertions have been generated by update_lit_checks.py --all-items and should not be edited.
-;; RUN: foreach %s %t wasm-opt --remove-unused-module-elements --closed-world --nominal -all -S -o - | filecheck %s
-;; RUN: foreach %s %t wasm-opt --remove-unused-module-elements                --nominal -all -S -o - | filecheck %s --check-prefix OPEN_WORLD
+;; RUN: foreach %s %t wasm-opt --remove-unused-module-elements --closed-world -all -S -o - | filecheck %s
+;; RUN: foreach %s %t wasm-opt --remove-unused-module-elements                -all -S -o - | filecheck %s --check-prefix OPEN_WORLD
 
 ;; Test both open world (default) and closed world. In a closed world we can do
 ;; more with function refs, as we assume nothing calls them on the outside, so
 ;; if no calls exist to the right type, the function is not reached.
 
 (module
-  ;; CHECK:      (type $A-super (func))
-  ;; OPEN_WORLD:      (type $A-super (func))
-  (type $A-super (func))
+  (rec
+    ;; CHECK:      (rec
+    ;; CHECK-NEXT:  (type $A-super (func))
+    ;; OPEN_WORLD:      (rec
+    ;; OPEN_WORLD-NEXT:  (type $A-super (func))
+    (type $A-super (func))
 
-  ;; CHECK:      (type $A (func_subtype $A-super))
-  ;; OPEN_WORLD:      (type $A (func_subtype $A-super))
-  (type $A (func_subtype $A-super))
+    ;; CHECK:       (type $A (func_subtype $A-super))
+    ;; OPEN_WORLD:       (type $A (func_subtype $A-super))
+    (type $A (func_subtype $A-super))
 
-  ;; CHECK:      (type $A-sub (func_subtype $A))
-  ;; OPEN_WORLD:      (type $A-sub (func_subtype $A))
-  (type $A-sub (func_subtype $A))
+    ;; CHECK:       (type $A-sub (func_subtype $A))
+    ;; OPEN_WORLD:       (type $A-sub (func_subtype $A))
+    (type $A-sub (func_subtype $A))
 
-  ;; CHECK:      (type $B (func))
-  ;; OPEN_WORLD:      (type $B (func))
-  (type $B (func))
+    ;; CHECK:       (type $B (func))
+    ;; OPEN_WORLD:       (type $B (func))
+    (type $B (func))
+  )
+
+  ;; CHECK:      (type $none_=>_none (func))
 
   ;; CHECK:      (elem declare func $target-A $target-A-sub $target-A-super $target-B)
 
   ;; CHECK:      (export "foo" (func $foo))
 
-  ;; CHECK:      (func $foo (type $A-super)
+  ;; CHECK:      (func $foo (type $none_=>_none)
```

Comment:
Why did this function's type change? That seems dangerous...

Reply:
There's no explicit type given in the input, so it was previously selecting the first compatible heap type, which happened to be `$A-super`. Now all the input types are in a nontrivial rec group, so none of them are eligible to be used as the function's type, so a new type is implicitly defined by the function definition.

I don't think this affects the intended purpose of the test in any way.